draft-wang-ppm-dap-taskprov.md

```diff
@@ -56,69 +60,302 @@ collector). Thus it is up to each deployment of DAP to securely implement its
 own mechanism.
 
 This document describes a mechanism for configuring tasks that may be useful in
-many deployments.
+many deployments. The goal of this mechanism is to add transparency to the
+task provision process, and define a task provision method that does not rely
+on deployment specific leader-helper out-of-band agreement.
+
+At a high level, this extension asks client to include all the task
+configuration parameters it received out-of-band from task author, in the
+extension field of the `Report` it uploads to aggregators. The aggregators will
+create a DAP task upon receiving `Report` with such extension (or `ReportShare`
+in the case of helper.)
+
+By sending task configuration parameters to clients, we add transparency to the
+task that clients participate in. Client can see what privacy parameters has
+been configured for a task (for e.g. `min_batch_size`, or any differential
+privacy parameters if that's the privacy guarantee used).
+
+By defining this mechanism as an extension, we can guarantee that for any
+deployments implementing this extension:
+
+1. Task transparency is enforced since aggregator task must be created using
+the same parameters client sent.
+
+2. No need for out-of-band task orchestration between leader and helpers, any
+leaders and helpers can work on the same DAP task, therefore making adoption of
+DAP easier.
+
+3. A "task" now means the same task ID and same task configuration, including
+parameters for choosing VDAF. Therefore, reports from one task will only be
+aggregated with one VDAF. For the same reason, if a malicious client changes
+the task ID or task configuration, its report will be aggregated in a different
+task, with other poison reports from the same malicious attack. The "good"
+reports will not be polluted.
 
-> TODO(wangshan) Describe the goals of this mechanism, including the deployment
-> scenario. Then describe how the extension works at a high level. I think the
-> intro should answer the following questions:
->
-> - Why does this mechanism need to be an extension (i.e., why are protocol
->   changes needed), as opposed to something that happens completely
->   out-of-band?
->
-> - Which of the sub-protocols are changed: upload, aggregate, collect?
+Because extension is used in HPKE's AAD, by including all task configurations
+in the extension, malicious leader cannot change the task configuration to
+mislead helpers to reduce privacy guarantee, for e.g. by reducing
+`min_batch_size`.
```

Comment:
This property as well should hold for any deployment of DAP. Since it's not particular to this extension, I don't think it's worth mentioning in the intro. Of course, in Security Considerations we'll want to discuss why this extension preserves the properties of DAP (i.e., does not introduce any regressions to security).

Reply:
let me move this to Security Considerations as a quoted comment, we can update/remove it when Security Considerations are fleshed out.